src/CadViewerManifold.tsx

```diff
@@ -223,42 +224,89 @@ try {
     }
   }, [])
 
+  // Check if we have a panel
+  const hasPanel = useMemo(
+    () => circuitJson.some((e) => e.type === "pcb_panel"),
+    [circuitJson],
+  )
+
+  // Use panel builder if panel exists, otherwise use board builder
+  const {
+    geoms: panelGeoms,
+    pcbThickness: panelThickness,
+    error: panelError,
+    isLoading: panelIsLoading,
+    panelData,
+  } = usePanelGeomBuilder(hasPanel ? manifoldJSModule : null, circuitJson)
+
   const {
     geoms,
     textures,
     pcbThickness,
     error: builderError,
     isLoading: builderIsLoading,
     boardData,
-  } = useManifoldBoardBuilder(manifoldJSModule, circuitJson)
-
-  const geometryMeshes = useMemo(() => createGeometryMeshes(geoms), [geoms])
-  const textureMeshes = useMemo(
-    () => createTextureMeshes(textures, boardData, pcbThickness),
-    [textures, boardData, pcbThickness],
-  )
+  } = useManifoldBoardBuilder(!hasPanel ? manifoldJSModule : null, circuitJson)
+
+  // Create meshes from panel or board geometry
+  const geometryMeshes = useMemo(() => {
+    if (hasPanel && panelGeoms?.panel) {
+      const mesh = new THREE.Mesh(
+        panelGeoms.panel.geometry,
+        new THREE.MeshStandardMaterial({
+          color: panelGeoms.panel.color,
+          roughness: 0.7,
+          metalness: 0.1,
+        }),
+      )
+      mesh.name = "panel-geom"
+      return [mesh]
+    }
+    return createGeometryMeshes(geoms)
+  }, [hasPanel, panelGeoms, geoms])
+  const textureMeshes = useMemo(() => {
+    if (hasPanel) return []
+    return createTextureMeshes(textures, boardData, pcbThickness)
+  }, [hasPanel, textures, boardData, pcbThickness])
 
   const cadComponents = useMemo(
     () => su(circuitJson).cad_component.list(),
     [circuitJson],
   )
 
   const boardDimensions = useMemo(() => {
+    if (hasPanel && panelData) {
+      return { width: panelData.width || 0, height: panelData.height || 0 }
+    }
     if (!boardData) return undefined
     const { width = 0, height = 0 } = boardData
     return { width, height }
-  }, [boardData])
+  }, [hasPanel, panelData, boardData])
 
   const boardCenter = useMemo(() => {
+    if (hasPanel && panelData?.center) {
+      return { x: panelData.center.x, y: panelData.center.y }
+    }
     if (!boardData) return undefined
     const { center } = boardData
     if (!center) return undefined
     return { x: center.x, y: center.y }
-  }, [boardData])
+  }, [hasPanel, panelData, boardData])
```

Comment:
See our code guidelines, “data” is a banned word, you can always find a better word

https://links.tscircuit.com